### ISSUE TITLE
Modeling the relationship between Category and Element

### ISSUE THREAD

**Issue by nicholaspoz** [UniversalScheduleStandard/UniversalScheduleStandard#2]
### Elements and Categories
In the mental model of the USS, `Categories` are a static list/taxonomy, and each `Element` belongs to exactly one `Category`. 

### Current Implementation
The current version of the USS models this relationship between by including an array of `Element` IDs inside the definition of the each `Category`. 
```js
{
  ...,
  elements: [{id: 'abc', ...}],
  categories: [{id: '123', elements: ['abc',...], ... }]
 }
```
Some of the drawbacks of this approach include
- The onus is on the developer to validate that every element is contained within exactly one category, which can be easily overlooked
- The act of changing the category of an element is a 2 step process: it must be removed from the old list and added to the new list
- It is slightly more work to see at a glance which category an element belongs to

### Suggested Change
It could be beneficial to model this relationship in the following way: 
- each `Element` has a required `categoryId` field
- remove the `elements` array from `Category`
```
{
  ...,
  elements: [{id: 'abc', categoryId: '123', ...}],
  categories: [{id: '123', ...}]
}
```

The same information is captured in this suggested model, but the rules of the relationship no longer require validation (each element belongs to exactly one category), and it is slightly easier to follow this relationship. The categories also become simpler, and look more like a static taxonomy which is in line with the mental model (and other models of taxonomies or "tags").

I hope this helps! Feedback on this suggestion is welcome, as well as additional context on how the current implementation came to be. 

Thanks for all the work that you do!
Nick @ Cinapse (nick@cinapse.io)

**Comment by modemmute** [comment]
This is an excellent suggestion, thank you for making it. 
I'm going to leave this issue open for a little while so that others may comment, but barring any disagreement, I think this suggestion is worthy of implementation. 
Does anyone disagree with the above suggestion or have any other ideas to further it?

**Comment by ldeboer** [comment]
Great suggestion, @nicholaspoz! I had the same thought as well, and would definitely +1 this suggestion.

**Comment by UniversalScheduleStandard** [comment]
This suggestion was accepted and is included in v0.9.1
